source/particle/generator/random_uniform.cc

```diff
@@ -0,0 +1,205 @@
+                  }
+              }
+
+            // Generate random points in these bounds until one is within the cell
+            unsigned int num_tries = 0;
+            while (num_tries < 100)
+              {
+                for (unsigned int d=0; d<dim; ++d)
+                  {
+                    pt[d] = uniform_distribution_01(random_number_generator) *
+                            (max_bounds[d]-min_bounds[d]) + min_bounds[d];
+                  }
+                try
+                  {
+                    const Point<dim> p_unit = this->get_mapping().transform_real_to_unit_cell(it, pt);
+                    if (GeometryInfo<dim>::is_inside_unit_cell(p_unit)) break;
+                  }
+                catch (typename Mapping<dim>::ExcTransformationFailed &)
+                  {
+                    // The point is not in this cell. Do nothing, just try again.
+                  }
+                num_tries++;
+              }
+            AssertThrow (num_tries < 100, ExcMessage ("Couldn't generate particle (unusual cell shape?)."));
+
+            // Add the generated particle to the set
+            const Particle<dim> new_particle(pt, cur_id);
+            particles.insert(std::make_pair(select_cell,new_particle));
+          }
+        return particles;
+      }
+
+
+      template <int dim>
+      void
+      RandomUniform<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Tracers");
+          {
+            prm.declare_entry ("Number of tracers", "1000",
+                               Patterns::Double (0),
+                               "Total number of tracers to create (not per processor or per element). "
+                               "The number is parsed as a floating point number (so that one can "
+                               "specify, for example, '1e4' particles) but it is interpreted as "
+                               "an integer, of course.");
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+
+
+      template <int dim>
+      void
+      RandomUniform<dim>::parse_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Tracers");
+          {
+            n_tracers    = static_cast<unsigned int>(prm.get_double ("Number of tracers"));
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      ASPECT_REGISTER_PARTICLE_GENERATOR(RandomUniform,
+                                         "random uniform",
+                                         "Generate random uniform distribution of "
+                                         "particles over entire simulation domain.")
```

Comment:
-> Generate a uniform random distribution of particles over the entire simulation domain